beamer/tests/agent/test_request.py

```diff
@@ -378,3 +378,20 @@ def test_event_fetcher_progress_without_events(agent, direction):
                 != ape.chain.blocks[-1].number
             ):
                 sleeper.sleep(0.1)
+
+
+def test_handling_reorgs(config, direction, token, request_manager):
+    requester, target = alloc_accounts(2)
+    confirmation_blocks = 1
+    config.chain_confirmation_blocks["l2a"] = confirmation_blocks
+    agent = Agent(config)
+    agent.start()
+
+    request_id = make_request(request_manager, token, requester, target, 1)
+    ape.chain.mine(confirmation_blocks)
+
+    with Sleeper(1) as sleeper:
+        while (agent.get_context(direction).requests.get(request_id)) is None:
+            sleeper.sleep(0.1)
+
+    agent.stop()
```

Comment:
How does this test work, i.e. how does a reorg happen and how do we check that we handled it correctly?

Reply:
so basically without extra ape.chain.mine(1) there we dont get the request at all, it is working in this way.

Reply:
so if request in block number 23 normally we can only reach it in block number 24 iterationn which we want

Reply:
I see, but the problem is, this test would be successful even without the confirmation blocks feature, right?. I.e. we are not handling a reorg _really_ because there is no reorg. We are just testing whether we wait for the specified number of confirmation blocks.

Reply:
isnt it enough? can you do a reorg in test?

Reply:
Well, one way to do it would be to 
1. make a snapshot
2. mine N blocks
3. roll back the chain to the snapshot
4. mine new blocks

I think that's as close as we could get to a real reorg with ape.
Now, for the actual test...

1. make a snapshot
2. mine N blocks, where `N > confirmation blocks`
3. make a request in another block, note the request ID, say `request1`
4. roll back the chain to the snapshot
5. mine N blocks
6. assert that the agent _did not_ see `request1`
7. make another request, note the request ID, say `request2`
8. mine `confirmation blocks` new blocks
9. assert that the agent _did_ see `request2`